admin/securedrop_admin/__init__.py

```diff
@@ -686,7 +686,18 @@ def update(args):
         if RELEASE_KEY in gpg_lines[1] and \
                 sig_result.count(good_sig_text) == 1 and \
                 bad_sig_text not in sig_result:
-            sdlog.info("Signature verification successful.")
+            # Finally, we check that there is no branch of the same name
+            # prior to reporting success.
+            cmd = ['git', 'show-ref', '--heads', '--verify', '--quiet',
+                   'refs/heads/{}'.format(latest_tag)]
+            try:
+                # We expect this to produce a non-zero exit code, which
+                # will produce a subprocess.CalledProcessError
+                subprocess.check_output(cmd)
+                sdlog.info("Signature verification failed.")
+                return 1
+            except subprocess.CalledProcessError:
```

Comment:
Are there any other circumstances/potential edge cases where this `CalledProcessError` exceptions can be raised? Would it be superfluous to verify the output text like in the admin integration tests (e.g. checking for the substring `not a valid ref`)?

Reply:
Hmm, I can't think of a reason off the top of my head, but for the sake of defense in depth, let me take a look at adding this (should be quick)

Reply:
this logic is added in e9998c5, good suggestion! also added 8fd25b0 for tests to cover the `else` case in the diff